app/src/main/java/by/lab/mypokemon/DetailsActivity.kt

```diff
@@ -17,27 +21,46 @@ class DetailsActivity : AppCompatActivity() {
         setContentView(binding.root)
 
         val pokemonId = intent.extras?.getInt("pokemon_id", -1) ?: -1
-        if (pokemonId != -1) {
+
+        with(binding) {
             val pokemon = PokemonRepository.getPokemonById(pokemonId)
-            if (pokemon != null) {
-                binding.pokemonImageView.setImageResource(pokemon.picture)
-                binding.pokemonNameTextView.text = pokemon.name
-                binding.pokemonWeightTextView.text = pokemon.weight.toString()
-                binding.pokemonHeightTextView.text = pokemon.height.toString()
-                val adapter = ArrayAdapter(this@DetailsActivity, android.R.layout.simple_list_item_1, pokemon.elementalType )
-                binding.pokemonTypeListView.adapter = adapter
-            } else {
-                Toast.makeText(this, "Unknown error. Try again later", Toast.LENGTH_SHORT).show()
+            pokemon?.let {
+                pokemonNameTextView.text = it.name
+                pokemonImageView.setImageResource(it.picture)
+                pokemonHeightTextView.text = "${it.height} cm"
+                pokemonWeightTextView.text = "${it.weight} kg"
+                createTypeListView(it.elementalType)
+            } ?: run {
                 Log.e("mainPage", "Pokemon with id not exist")
+                Toast.makeText(this@DetailsActivity, "Unknown error. Try again later", Toast.LENGTH_SHORT).show()
                 finish()
             }
-        } else {
-            Toast.makeText(this, "Unknown error. Try again later", Toast.LENGTH_SHORT).show()
-            Log.e("mainPage", "Invalid id")
-            finish()
         }
-        binding.backBtn.setOnClickListener{
+        binding.backBtn.setOnClickListener {
             finish()
         }
     }
+
+    private fun createTypeListView(typeList: List<String>) {
+        var prevViewId = binding.typeView.id
+        typeList.forEach {
+            val textView = TextView(this).apply {
+                text = it
+                setTextSize(TypedValue.COMPLEX_UNIT_SP, 24F)
```

Comment:
используй сеттер, он по умолчанию принимает sp
```suggestion
                textSize = 24f
```